[ews-build.webkit.org] Treat _addLog as asynchronous in ValidateChange

JonWBedard opened the pull request:
#### 1c9d439b0c6bfee4029253c952738aa937bd2894
<pre>
[ews-build.webkit.org] Treat _addLog as asynchronous in ValidateChange
<a href="https://bugs.webkit.org/show_bug.cgi?id=251632">https://bugs.webkit.org/show_bug.cgi?id=251632</a>
rdar://104977612

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(GitHubMixin._is_pr_closed): Convert to asynchronous function.
(GitHubMixin._is_hash_outdated): Ditto.
(BugzillaMixin._is_patch_obsolete): Ditto.
(BugzillaMixin._is_patch_review_denied): Ditto.
(BugzillaMixin._is_patch_cq_plus): Ditto.
(BugzillaMixin._does_patch_have_acceptable_review_flag): Ditto.
(BugzillaMixin._is_bug_closed): Ditto.
(ValidateChange.validate_bugzilla): Treat many functions as asynchronous.
(ValidateChange.validate_github): Ditto.
(ValidateChange.send_email_for_github_failure): Convert to asynchronous function.

Canonical link: <a href="https://commits.webkit.org/259782@main">https://commits.webkit.org/259782@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/95ae3e4cb33eca328f032dce1f68d87e51691864

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/105936 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/15002 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/38780 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/115125 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/175243 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/11/builds/109837 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/16421 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/85/builds/6188 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/98159 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/114876 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/111686 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/12496 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/95465 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/98159 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/94361 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/27109 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/98159 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/8262 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/28462 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/8747 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/85/builds/6188 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/104738 "Passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/14375 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/48004 "Passed tests") | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/79/builds/10297 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3628 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

aj062 reviewed (approved):
r+ assuming this is tested

JonWBedard committed:
[ews-build.webkit.org] Treat _addLog as asynchronous in ValidateChange
https://bugs.webkit.org/show_bug.cgi?id=251632
rdar://104977612

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(GitHubMixin._is_pr_closed): Convert to asynchronous function.
(GitHubMixin._is_hash_outdated): Ditto.
(BugzillaMixin._is_patch_obsolete): Ditto.
(BugzillaMixin._is_patch_review_denied): Ditto.
(BugzillaMixin._is_patch_cq_plus): Ditto.
(BugzillaMixin._does_patch_have_acceptable_review_flag): Ditto.
(BugzillaMixin._is_bug_closed): Ditto.
(ValidateChange.validate_bugzilla): Treat many functions as asynchronous.
(ValidateChange.validate_github): Ditto.
(ValidateChange.send_email_for_github_failure): Convert to asynchronous function.

Canonical link: https://commits.webkit.org/259782@main
